server/src/main/java/io/crate/execution/engine/FirstColumnConsumers.java

```diff
@@ -21,51 +21,51 @@
 
 package io.crate.execution.engine;
 
-import io.crate.data.CollectingRowConsumer;
-import io.crate.data.Row;
-
 import java.util.ArrayList;
-import java.util.Collection;
 import java.util.Collections;
+import java.util.List;
 import java.util.Set;
 import java.util.function.BiConsumer;
 import java.util.function.BinaryOperator;
 import java.util.function.Function;
 import java.util.function.Supplier;
 import java.util.stream.Collector;
 
+import io.crate.data.Row;
+import io.crate.expression.symbol.SelectSymbol.ResultType;
+
 /**
  * Collectors to retrieve either {@link AllValues} or a {@link SingleValue} of the first column of each row.
  */
 public class FirstColumnConsumers {
 
-    private static class AllValues implements Collector<Row, Collection<Object>, Object[]> {
+    private static class AllValues implements Collector<Row, List<Object>, List<Object>> {
```

Comment:
I'm not entirely sure why this wasn't a problem before. Arrays are supposed to be of type `List<?>`, not `Object[]`. My best guess is that IN(..) and = ANY(...)  expressions run it through `ArrayType.value(..)` which casts `Object[]` to `List<Object>`, but I didn't bother validating - I couldn't trigger a user facing bug with the previous implementation.